GameServerLib/Maps/SummonersRift.cs

```diff
@@ -247,135 +312,262 @@ public void Init()
-            _purpleInhibitors.Add(new Inhibitor(_game, "ChaosInhibitor", TeamId.TEAM_PURPLE, inhibRadius, new Vector2(11196.097f, 11439.8077f), sightRange, 0xffff8f1f));
-            _purpleInhibitors.Add(new Inhibitor(_game, "ChaosInhibitor", TeamId.TEAM_PURPLE, inhibRadius, new Vector2(13196.097f, 11164.8077f), sightRange, 0xff26ac0f));
-
-            // Nexus turrets
-            _purpleNexusTurrets.Add(new LaneTurret(_game, "Turret_T2_C_01_A", "ChaosTurretNormal", new Vector2(12621.097f, 12364.8077f), TeamId.TEAM_PURPLE,
-                TurretType.NEXUS_TURRET, GetTurretItems(TurretType.NEXUS_TURRET)));
-            _purpleNexusTurrets.Add(new LaneTurret(_game, "Turret_T2_C_02_A", "ChaosTurretNormal", new Vector2(12171.097f, 12789.8077f), TeamId.TEAM_PURPLE,
-                TurretType.NEXUS_TURRET, GetTurretItems(TurretType.NEXUS_TURRET)));
-
-            // Fountain turrets
-            _game.ObjectManager.AddObject(new LaneTurret(_game, "Turret_OrderTurretShrine_A", "OrderTurretShrine", new Vector2(-236.05f, -53.32f), TeamId.TEAM_BLUE,
-                TurretType.FOUNTAIN_TURRET, GetTurretItems(TurretType.FOUNTAIN_TURRET)));
-            _game.ObjectManager.AddObject(new LaneTurret(_game, "Turret_ChaosTurretShrine_A", "ChaosTurretShrine", new Vector2(14157.0f, 14456.0f), TeamId.TEAM_PURPLE,
-                TurretType.FOUNTAIN_TURRET, GetTurretItems(TurretType.FOUNTAIN_TURRET)));
-
-            // TODO: Decide if we want to let ObjectManager handle protection automatically.
-            for (int i = 0; i < _blueOuterTurrets.Count; i++)
-            {
-                _game.ProtectionManager.AddProtection(_purpleInhibitors[i], false, _purpleInhibTurrets[i]);
-                _game.ProtectionManager.AddProtection(_purpleInhibTurrets[i], false, _purpleInnerTurrets[i]);
-                _game.ProtectionManager.AddProtection(_purpleInnerTurrets[i], false, _purpleOuterTurrets[i]);
-
-                _game.ProtectionManager.AddProtection(_blueInhibitors[i], false, _blueInhibTurrets[i]);
-                _game.ProtectionManager.AddProtection(_blueInhibTurrets[i], false, _blueInnerTurrets[i]);
-                _game.ProtectionManager.AddProtection(_blueInnerTurrets[i], false, _blueOuterTurrets[i]);
+            // Below is where we create the buildings.
+
+            // These two are used for fixing any wrongly indexed turrets that are present.
+            List<MapData.MapObject> missedTurrets = new List<MapData.MapObject>();
+
+            // This is all of the possible inhibitor NetIDs, in the order of bot-mid-top, with 0 -> 2 being Order, and 3 -> 5 being Chaos.
+            // Figure out why they hardcoded the NetIDs (and if we can work around it, do so to maximize moddability).
+            uint[] inhibitorNetIDs = new uint[6] { 0xff9303e1, 0xff4a20f1, 0xffd23c3e, 0xff26ac0f, 0xffff8f1f, 0xff6793d0 };
```

Comment:
If I remember correctly these net ids are generated from constant string (something like 0xff000000 or something similar) and the hash (there are hash functions somewhere in the project) of the inhibitor name. I'll add it as an issue.

Reply:
These net ids are generated as follows:
CRC32(original barracks name - for example "Barracks_T1_L1","Barracks_T2_C1","Barracks_T1_R1") | 0xFF000000

See BaseTurret.cs for the CRC32 lib in C# that we are already using.